### DISCUSSION TITLE
Is it possible to handle CSS in the static site generator?

### DISCUSSION THREAD

**Discussion by richi** [datenstrom/community#145] [Ask a question] [answered]
I have just build a small CSS preprocessor (SCSS style variables) as a plugin:

https://github.com/richi/yellowcms-dynamiccss

Unfortunately I had to change a few lines in commandline.php to get it to work during static builds. Right now, only a `clean`command is routed to plugins during a build.

line 217:

```
function buildStaticFile($fileNameSource, $fileNameDest, $fileTypeMedia)
{
    $statusCode = $this->yellow->toolbox->copyFile($fileNameSource, $fileNameDest, true) &&
            $this->yellow->toolbox->modifyFile($fileNameDest, filemtime($fileNameSource)) ? 200 : 500;
```

changed to:

```
function buildStaticFile($fileNameSource, $fileNameDest, $fileTypeMedia)
{
    $statusCode = $this->pluginCommand(array($fileNameSource, "buildStatic", $fileNameDest, $fileTypeMedia));
    if ($statusCode == 0)
    {
        $statusCode = $this->yellow->toolbox->copyFile($fileNameSource, $fileNameDest, true) &&
            $this->yellow->toolbox->modifyFile($fileNameDest, filemtime($fileNameSource)) ? 200 : 500;
    }   
```

Does that look OK? And could we get a hook like this in the next release? Pretty please :)

Cheers, richi

**Comment by markseuffert** [comment]
Thanks for the suggestions. I am afraid to say that we don't add a command. We can't promise if we may add something like this, however we definitely appreciate the feedback.

Thanks, Mark

**Comment by markseuffert** [comment, marked as answer]
Have a look at latest version, you should get access to CSS files via `onRequest()`.

**Comment by richi** [comment]
Perfect, works great!
